crates/oxc_linter/src/ast_util.rs

```diff
@@ -381,3 +381,29 @@ pub fn get_new_expr_ident_name<'a>(new_expr: &'a NewExpression<'a>) -> Option<&'
 
     Some(ident.name.as_str())
 }
+
+/// Check if the given [IdentifierReference] is a global reference.
+/// Such as `window`, `document`, `globalThis`, etc.
+pub fn is_global_reference(
+    ident: &IdentifierReference,
+    ctx: &LintContext,
+) -> bool {
+    let symbol_table = ctx.semantic().symbols();
+    let Some(reference_id) = ident.reference_id.get() else {
+        return false;
+    };
+    let reference = symbol_table.get_reference(reference_id);
+    reference.symbol_id().is_none()
+}
+
+pub fn is_global_require_call(call_expr: &CallExpression, ctx: &LintContext) -> bool {
+    if call_expr.arguments.len() != 1 {
+        return false;
+    }
+
+    if let Expression::Identifier(id_ref) = &call_expr.callee {
+        id_ref.name == "require" && is_global_reference(id_ref, ctx)
```

Comment:
I remove some of validations

https://github.com/oxc-project/oxc/blob/ef336cb66b4746073e9d6a2f950d46185fec3cf6/crates/oxc_ast/src/ast/js.rs#L728-L730

I guess even users use `require` in a wrong way, we probably still want to catch them in this rule.